diff --git a/src/imf.cpp b/src/imf.cpp
index 4971bda..5ecfd81 100644
--- a/src/imf.cpp
+++ b/src/imf.cpp
@@ -89,7 +89,7 @@ bool CimfPlayer::load(const std::string &filename, const CFileProvider &fp)
   else
     fsize = f->readInt(2);
   flsize = fp.filesize(f);
-  if (mfsize + 4 > flsize || fsize >= flsize - 2 - mfsize || fsize & 3) {
+  if (mfsize + 4 > flsize || fsize > flsize - 2 - mfsize || fsize & 3) {
     fp.close(f);        // truncated file or bad size record
     return false;
   }    
diff --git a/src/version.h b/src/version.h
new file mode 100644
index 0000000..86ec36f
--- /dev/null
+++ b/src/version.h
@@ -0,0 +1 @@
+#define ADPLUG_VERSION "2.3.3"
