diff --git a/src/flowgraph/resampler/PolyphaseResampler.cpp b/src/flowgraph/resampler/PolyphaseResampler.cpp
index 4862ad8..163cf7c 100644
--- a/src/flowgraph/resampler/PolyphaseResampler.cpp
+++ b/src/flowgraph/resampler/PolyphaseResampler.cpp
@@ -14,6 +14,7 @@
  * limitations under the License.
  */
 
+#include <cassert>
 #include <math.h>
 #include "IntegerRatio.h"
 #include "PolyphaseResampler.h"
diff --git a/src/flowgraph/resampler/PolyphaseResamplerMono.cpp b/src/flowgraph/resampler/PolyphaseResamplerMono.cpp
index 2dcdc8e..c0e29b7 100644
--- a/src/flowgraph/resampler/PolyphaseResamplerMono.cpp
+++ b/src/flowgraph/resampler/PolyphaseResamplerMono.cpp
@@ -14,6 +14,7 @@
  * limitations under the License.
  */
 
+#include <cassert>
 #include "PolyphaseResamplerMono.h"
 
 using namespace resampler;
diff --git a/src/flowgraph/resampler/PolyphaseResamplerStereo.cpp b/src/flowgraph/resampler/PolyphaseResamplerStereo.cpp
index 5c73a8e..e4bef74 100644
--- a/src/flowgraph/resampler/PolyphaseResamplerStereo.cpp
+++ b/src/flowgraph/resampler/PolyphaseResamplerStereo.cpp
@@ -14,6 +14,7 @@
  * limitations under the License.
  */
 
+#include <cassert>
 #include "PolyphaseResamplerStereo.h"
 
 using namespace resampler;
diff --git a/src/flowgraph/resampler/SincResampler.cpp b/src/flowgraph/resampler/SincResampler.cpp
index 1084356..3758210 100644
--- a/src/flowgraph/resampler/SincResampler.cpp
+++ b/src/flowgraph/resampler/SincResampler.cpp
@@ -14,6 +14,7 @@
  * limitations under the License.
  */
 
+#include <cassert>
 #include <math.h>
 #include "SincResampler.h"
 
diff --git a/src/flowgraph/resampler/SincResamplerStereo.cpp b/src/flowgraph/resampler/SincResamplerStereo.cpp
index bde658a..ce00302 100644
--- a/src/flowgraph/resampler/SincResamplerStereo.cpp
+++ b/src/flowgraph/resampler/SincResamplerStereo.cpp
@@ -14,6 +14,7 @@
  * limitations under the License.
  */
 
+#include <cassert>
 #include <math.h>
 
 #include "SincResamplerStereo.h"
